src/core/lib/security/transport/security_connector.c

```diff
@@ -637,6 +686,26 @@ static void ssl_server_check_peer(grpc_exec_ctx *exec_ctx,
   GRPC_CLOSURE_SCHED(exec_ctx, on_peer_checked, error);
 }
 
+static int ssl_channel_cmp(grpc_security_connector *sc1,
+                           grpc_security_connector *sc2) {
+  grpc_ssl_channel_security_connector *c1 =
+      (grpc_ssl_channel_security_connector *)sc1;
+  grpc_ssl_channel_security_connector *c2 =
+      (grpc_ssl_channel_security_connector *)sc2;
+  int c = grpc_channel_security_connector_cmp(&c1->base, &c2->base);
+  if (c != 0) return c;
```

Comment:
@justinburke FYI.
Looks like key, cert and roots are missing here... which is a potential security issue.
We'll probably need a comparison function for the handshaker factory.

Reply:
I've added incomplete comparison functions for the SSL handshaker factories.  The thing that's still missing is a way to compare the SSL_CTX objects, and I'm hoping you can help me with that, because I'm not very familiar with the OpenSSL API.  Is there an easy way to do this?  I did a quick websearch but didn't come up with anything, so I'm hoping you can suggest something.  The only alternative I can think of is to save our own copy of the parameters that we use to construct the SSL_CTX object, but that seems like a waste of memory, so I'm hoping that there's a better way.